opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1492,6 +1492,8 @@ namespace Opm {
             auto& events = this->wellState().well(well->indexOfWell()).events;
             if (events.hasEvent(WellState::event_mask)) {
                 well->updateWellStateWithTarget(ebosSimulator_, this->groupState(), this->wellState(), deferred_logger);
+                well->updatePrimaryVariables(this->wellState(), deferred_logger);
```

Comment:
Basically there is no updating of the primary variables before the `solveWellEquation`. Difficult to see the impact of it.